### PR TITLE
Refactor(Dispatcher): Refactoring Dispatcher, UoW

### PR DESCRIPTION
**Pull request by zhenlei520**
# Description

Refactor Dispatcher, UoW, fix the problem of event sequence sent by aggregate root

## Checklist

Please make sure you've completed the relevant tasks for this PR, out of the following list:

* [x] Code compiles correctly
* [x] Created/updated tests
* [x] Extended the documentation

### CODE REVIEW

File: src/Dispatcher/Masa.BuildingBlocks.Dispatcher.Events/IMiddleware.cs [hunk 1]
```diff
@@ -9,7 +9,13 @@ namespace Masa.BuildingBlocks.Dispatcher.Events;
...
 /// </summary>
 public interface IMiddleware<TEvent>
-    where TEvent : notnull, IEvent
+    where TEvent : IEvent
 {
     Task HandleAsync(TEvent @event, EventHandlerDelegate next);
+
+    /// <summary>
+    /// If looping is not supported, the current Middleware only executes once
+    /// If looping is supported, Middleware will be executed multiple times when EventBus is nested
```
Comment (doddgu) [inline src/Dispatcher/Masa.BuildingBlocks.Dispatcher.Events/IMiddleware.cs:18]: multiple? everytime?

**Review by doddgu — CHANGES_REQUESTED** [review]

File: src/Dispatcher/Masa.BuildingBlocks.Dispatcher.Events/IMiddleware.cs (hunk 1 above)
Comment (zhenlei520) [reply src/Dispatcher/Masa.BuildingBlocks.Dispatcher.Events/IMiddleware.cs:18]: Processed

File: src/Dispatcher/Masa.BuildingBlocks.Dispatcher.Events/IMiddleware.cs [hunk 2]
```diff
@@ -9,7 +9,13 @@ namespace Masa.BuildingBlocks.Dispatcher.Events;
 /// Middleware is assembled into an event pipeline to handle invoke event and result
 /// </summary>
 public interface IMiddleware<TEvent>
-    where TEvent : notnull, IEvent
+    where TEvent : IEvent
 {
     Task HandleAsync(TEvent @event, EventHandlerDelegate next);
+
+    /// <summary>
+    /// If looping is not supported, the current Middleware only executes once
```
[the remark this answers is not in the document]
Comment (zhenlei520) [reply src/Dispatcher/Masa.BuildingBlocks.Dispatcher.Events/IMiddleware.cs:17]: Processed

**Review by doddgu — APPROVED** [review]
LGTM

**Pull request merged**